crates/server/src/authority/catalog.rs

```diff
@@ -634,26 +638,99 @@ async fn send_authoritative_response(
                 }
             }
         } else {
+            #[cfg(feature = "dnssec")]
+            {
+                if let Some(NxProofKind::Nsec3 {
+                    algorithm,
+                    salt,
+                    iterations,
+                }) = authority.nx_proof_kind()
+                {
+                    // This unwrap will not panic as we know that `answers` is `Some`.
+                    let has_wildcard_match =
+                        answers.as_ref().unwrap().iter().any(|rr| {
+                            rr.record_type() == RecordType::RRSIG && rr.name().is_wildcard()
+                        });
+                    match authority
+                        .get_nsec3_records(
+                            Nsec3QueryInfo {
+                                qname: query.name(),
+                                qtype: query.query_type(),
+                                has_wildcard_match,
+                                algorithm: *algorithm,
+                                salt,
+                                iterations: *iterations,
+                            },
+                            lookup_options,
+                        )
+                        .await
+                    {
+                        // run the soa lookup
+                        Continue(Ok(nsecs)) | Break(Ok(nsecs)) => (Some(nsecs), None),
+                        Continue(Err(e)) | Break(Err(e)) => {
+                            warn!("failed to lookup nsecs: {}", e);
```

Comment:
can we add the request_id to this? seems like that would be more useful on the warning.